src/Altinn.AccessManagement.Persistence/DelegationMetadataRepository.cs

```diff
@@ -5,525 +5,644 @@
+        catch (Exception e)
+        {
+            _logger.LogError(e, "Authorization // DelegationMetadataRepository // GetReceivedResourceRegistryDelegationsForCoveredByParty // Exception");
+            throw;
+        }
+    }
 
-        /// <inheritdoc/>
-        public async Task<List<DelegationChange>> GetReceivedResourceRegistryDelegationsForCoveredByUser(int coveredByUserId, List<int> offeredByPartyIds, List<string> resourceRegistryIds = null, List<ResourceType> resourceTypes = null)
+    /// <inheritdoc/>
+    public async Task<List<DelegationChange>> GetAllDelegationChangesForAuthorizedParties(List<int> coveredByUserIds, List<int> coveredByPartyIds, CancellationToken cancellationToken = default)
+    {
+        if (coveredByUserIds == null && coveredByPartyIds == null)
         {
-            try
-            {
-                await using NpgsqlConnection conn = new NpgsqlConnection(_connectionString);
-                await conn.OpenAsync();
-
-                NpgsqlCommand pgcom = new NpgsqlCommand(getResourceRegistryDelegationChangesForCoveredByUserId, conn);
-                pgcom.Parameters.AddWithValue("_coveredByUserId", coveredByUserId);
-                pgcom.Parameters.AddWithValue("_offeredByPartyIds", NpgsqlTypes.NpgsqlDbType.Array | NpgsqlTypes.NpgsqlDbType.Integer, (offeredByPartyIds == null || !offeredByPartyIds.Any()) ? DBNull.Value : offeredByPartyIds);
-                pgcom.Parameters.AddWithValue("_resourceRegistryIds", NpgsqlTypes.NpgsqlDbType.Array | NpgsqlTypes.NpgsqlDbType.Text, (resourceRegistryIds == null || !resourceRegistryIds.Any()) ? DBNull.Value : resourceRegistryIds);
-                pgcom.Parameters.AddWithValue("_resourceTypes", NpgsqlTypes.NpgsqlDbType.Array | NpgsqlTypes.NpgsqlDbType.Text, (resourceTypes == null || !resourceTypes.Any()) ? DBNull.Value : resourceTypes.Select(rt => rt.ToString().ToLower()).ToList());
-
-                List<DelegationChange> receivedDelegations = new List<DelegationChange>();
-                using NpgsqlDataReader reader = pgcom.ExecuteReader();
-                while (reader.Read())
-                {
-                    receivedDelegations.Add(GetResourceRegistryDelegationChange(reader));
-                }
-
-                return receivedDelegations;
-            }
-            catch (Exception e)
-            {
-                _logger.LogError(e, "Authorization // DelegationMetadataRepository // GetReceivedResourceRegistryDelegationsForCoveredByParty // Exception");
-                throw;
-            }
+            return await Task.FromResult(new List<DelegationChange>());
         }
 
-        private async Task<DelegationChange> InsertAppDelegation(DelegationChange delegationChange)
+        const string QUERY = /*strpsql*/@"
+            WITH resources AS (
+		        SELECT
+			        resourceId,
+			        resourceRegistryId,
+			        resourceType
+		        FROM accessmanagement.Resource
+		        WHERE resourceType != 'maskinportenschema'
```

Comment:
case sensitive?

Reply:
safer to have this value as arg maybe?

Reply:
I think I prefer that we start to make queries for their specific intended use instead of the more generic approach with multiple input arguments. But as we get multiple queries which will use the same filtering we can/should make some of these reusable across queries.

I also don't want to do too much in the DelegationMetadataRepository now since @Thuen is doing a major rewrite